interval/search_test.go

```diff
@@ -587,9 +590,63 @@ func TestSearchTree_Select(t *testing.T) {
 	}
 }
 
+func TestSearchTree_EncodingDecoding(t *testing.T) {
```

Comment:
This test encodes a tree and decodes it in another instance.
Then, it compares them to make sure they have the same records.

Reply:
Do you mind moving this tests to the `search_tree_test.go` , as the implementation of the `encoding/gob` interfaces is not part of the `search.go` file.